new/ohai-cookbook-segment.md

```diff
@@ -0,0 +1,45 @@
+---
+RFC: unassigned
+Author: Lamont Granquist <lamont@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Title
+
+Support ohai plugins under an `ohai` top level directory in cookbooks.  Load all
+ohai plugins in all synchronized cookbooks after cookbook synchronization.
+
+## Motivation
+
+    As a Chef User,
+    I want to have my custom Ohai plugins loaded on first bootstrap,
+    So I don't have to run chef twice.
+
+    As a Chef User,
+    I want my Ohai plugins loaded before attributes and compile/converge mode,
+    So I can use them without worrying about cookbook execution ordering.
+
+    As a Chef User,
+    I want my Ohai plugins synchronized with my cookbooks,
+    So that I don't incure more unavoidable round-trips to the Chef Server.
+
+    As a Chef Developer,
+    I want Ohai plugins as a first-class object,
+    So that I don't have to compile recipes to discover templates that drop plugins.
+
+## Specification
+
+The "segments" of a cookbook will be extended to include an "ohai" segment.  In this
+segment there will be plugins which are intended to be copied to the Ohai `plugin_path`.  All files in this segment will be copied, recursively, maintaining directory
+structure.
```

Comment:
If we overwrite, we can workaround issues with existing core plugins temporarily. For instance, in https://github.com/chef/ohai/issues/420 a user is using an old version of hostname.rb to get back to older behavior that worked for them. 

Of course, I'd want to warn to keep someone from doing that forever or it happening unexpectedly, but then I'd want to be able to turn off the warning if my plugin replacement was never going upstream. Which is all a tiring pattern.